scene/main/window.cpp

```diff
@@ -32,11 +32,16 @@
 
 #include "core/config/project_settings.h"
 #include "core/debugger/engine_debugger.h"
+#include "core/error/error_macros.h"
 #include "core/input/shortcut.h"
+#include "core/io/config_file.h"
 #include "core/string/translation_server.h"
+#include "core/string/ustring.h"
+#include "editor/editor_paths.h"
```

Comment:
Files in `scene/` should not include editor-only code, even behind `#ifdef TOOLS_ENABLED`. You need to find a different way to expose this (perhaps in a way that projects can use, so this can also address https://github.com/godotengine/godot-proposals/issues/10625).